### PR TITLE
App Lab: make it possible to edit start html in the editor

### PR DESCRIPTION
**Pull request by joshlory**
Request from @gtw and @lindyroo to make the start html and code directly editable in place, rather than needing to copy & paste the code from the console of a different tab.

### CODE REVIEW

**Comment by joshlory** [comment]
To edit starter code/html for an App Lab level:

image

Which takes you to this page:

image

Clicking "Save" will update the starter code and html for the level. The normal edit text areas on the level edit page will continue to work (and should show the newly-saved starter code/html after reloading).

File: apps/src/code-studio/header.js [hunk 1]
```diff
@@ -351,6 +366,24 @@ header.showMinimalProjectHeader = function() {
   $('.project_remix').click(remixProject);
 };
 
+// Levelbuilder-only UI for saving changes to a level.
+header.showLevelBuilderSaveButton = function(getChanges) {
+  var projectName = $('<div class="project_name_wrapper header_text">')
```
Comment (joshlory) [inline apps/src/code-studio/header.js:371]: This header UI is still in jQuery 😢. I opened https://codedotorg.atlassian.net/browse/LP-131 to track.

**Comment by islemaster** [comment]
Just starting to review, but wanted to throw in: :100: on taking over the site header space for superuser features.  I'd love to have a whole header mode where it changes color and plays host to all of our various support-staff powers and debug menus.

File: dashboard/app/controllers/levels_controller.rb [hunk 2]
```diff
@@ -100,7 +100,22 @@ def update_blocks
     @level.properties[type] = blocks_xml
     @level.log_changes(current_user)
     @level.save!
-    render json: {redirect: level_url(@level)}
+    render json: {redirect: edit_level_url(@level)}
```
Comment (islemaster) [inline dashboard/app/controllers/levels_controller.rb]: Does this change the blockly editing workflow to send you to the level edit page every time you change start blocks?  That seems potentially annoying if you're iterating on something; unless I missed this, and you've unhooked the save-on-run thing.

**Review by islemaster — APPROVED** [review]
:tada: :rocket: Nice!

File: dashboard/app/controllers/levels_controller.rb (hunk 2 above)
Comment (joshlory) [reply dashboard/app/controllers/levels_controller.rb]: You're right, I should leave this as-is. I'll change it back!

**Pull request merged**